src/autowiring/test/AutoConstructTest.cpp

```diff
@@ -102,3 +102,46 @@ TEST_F(AutoConstructTest, FactoryNewPrivateCtor) {
   ASSERT_NE(nullptr, mpcc.get()) << "Null not expected as a return type from a factory new construction";
   ASSERT_EQ(1002, mpcc->ival) << "Correct ctor was not invoked on a type with a private ctor";
 }
+
+namespace {
+  class MyPrivateCtorStringClass :
+    public CoreObject
+  {
+    MyPrivateCtorStringClass(void) :
+      istr("default_string")
+    {}
+    MyPrivateCtorStringClass(const char*) :
+      istr(istr)
```

Comment:
The member variable`istr` ends up being uninitialized here, as you are ignoring the parameter passed to the ctor.